### PR TITLE
Line map pull request

### PR DESCRIPTION
**Pull request by debjyoti0891**
1. updated the step data structure
2. line_map is now a member of the structure
3. updated synthesize_node function to take line map as input.

Passes for :
lhrs -k 6 -m 0 --esopscript 0 --esopcovermethod bdd
lhrs -k 6 -m 0 --esopscript 0 --esopcovermethod auto
lhrs -k 6 -m 0 --esopscript 0 --esopcovermethod aignew
lhrs -k 6 -m 0 --esopscript 0 --esopcovermethod aig

write_qc fails :
for other methods(-m).

### CODE REVIEW

File: addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp [hunk 1]
```diff
@@ -108,6 +108,7 @@ class lut_order_heuristic
     unsigned              target;        /* the target line for the result */
     step_type             type;          /* which step to perform */
     std::vector<unsigned> clean_ancilla; /* number of clean ancillae */
+    std::vector<unsigned> line_map;
```
Comment (msoeken) [inline addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp]: Please put a comment like for the other fields.

File: addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp [hunk 2]
```diff
@@ -183,7 +184,10 @@ class lut_order_heuristic
   {
     if ( !_dry_run )
     {
-      _steps.push_back( {index, target, type, _constants} );
+      if(type == step_type::compute || type == step_type :: uncompute)
```
Comment (msoeken) [inline addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp:187]: Please follow coding style for spaces around expressions, `if`, indentation, and use of braces for all blocks.

File: addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp [hunk 3]
```diff
@@ -517,14 +521,14 @@ class lut_based_synthesis_manager
...
-          synthesize_node( step.node, false, step.clean_ancilla );
+          synthesize_node( step.node, false, step.clean_ancilla, step.line_map );
         }
         break;
 
       case lut_order_heuristic::uncompute:
         if ( !params.onlylines )
         {
-          synthesize_node( step.node, true, step.clean_ancilla );
+          synthesize_node( step.node, true, step.clean_ancilla , step.line_map);
```
Comment (msoeken) [inline addons/cirkit-addon-reversible/src/reversible/synthesis/lhrs/legacy/lhrs.cpp]: Wrong spaces.

**Comment by msoeken** [comment]
Hi, thanks for the pull request. What do you mean by

> write_qc fails

**Comment by debjyoti0891** [comment]
Please see our Skype conversation. I pasted the exact error.

**Comment by msoeken** [comment]
Thanks for the pull request. I fix the remaining indentation issues myself.

**Pull request merged**